### ISSUE TITLE
Crash if missing parameter in non-tty environment without indication which parameter is missing

### ISSUE THREAD

**Issue by liamdawson** [Azure/azure-cli#3876]
### Description

If a parameter is missing when doing a resource group template deploy, `az` attempts to prompt, but crashes upon finding it's not in a TTY, and therefore does not explain which parameter is missing. This means it is not possible to see which parameter is missing when run in a CI environment, for example.

---

### Environment summary

**Install Method:** `pip`

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
```
azure-cli (2.0.2)

acr (2.0.0)
acs (2.0.2)
appservice (0.1.2)
batch (2.0.0)
cloud (2.0.0)
component (2.0.0)
configure (2.0.2)
container (0.1.2)
core (2.0.2)
documentdb (0.1.2)
feedback (2.0.0)
find (0.0.1b1)
iot (0.1.2)
keyvault (2.0.0)
lab (0.0.1)
monitor (0.0.1)
network (2.0.2)
nspkg (2.0.0)
profile (2.0.2)
redis (0.1.1b3)
resource (2.0.2)
role (2.0.1)
sql (2.0.0)
storage (2.0.2)
vm (2.0.2)

Python (Linux) 2.7.12 (default, Nov 19 2016, 06:48:10)
[GCC 5.4.0 20160609]
```

**OS Version:** Ubuntu 16.04, in Docker

**Shell Type:** bash

**Comment by derekbekoe** [comment]
@liamdawson What command was attempted?

**Comment by liamdawson** [comment]
`az group deployment create`.

For us, the trigger issue was actually that the syntax had changed which meant the parameters file was being used incorrectly (`--parameters '@file.json'` -> `--parameters 'file.json'`), but due to this issue it took some debugging as we normally run the task from a CI environment.

**Comment by derekbekoe** [comment]
This may have been fixed by https://github.com/Azure/azure-cli/pull/3850 that was recently merged.
@tjprescott can you comment?

**Comment by tjprescott** [comment]
Actually, this was most likely fixed by #3743. Either syntax (--parameters @file.json and --parameters file.json) should work and were intended to work.  However, this doesn't really address the underlying issue which is that the NoTTYError doesn't give any indication of what was being prompted for.
